layers/+source-control/git/packages.el

```diff
@@ -30,10 +30,17 @@
         ))
 
 (defun git/init-evil-magit ()
+  ;; Enabled `evil-magit' so that switching between editing styles keeps the
+  ;; keybindings consistent
   (with-eval-after-load 'magit
     (require 'evil-magit)
     (evil-define-key 'motion magit-mode-map
-      (kbd dotspacemacs-leader-key) spacemacs-default-map)))
+      (kbd dotspacemacs-leader-key) spacemacs-default-map)
+    ;; `dotspacemacs-editing-style-hook' can't take care of deferred loading of
+    ;; `evil-magit', hence disable it here for `emacs' style
+    (when (eq dotspacemacs-editing-style 'emacs)
+      (evil-magit-revert)))
+  (add-hook 'spacemacs-editing-style-hook 'spacemacs//magit-evil-magit-bindings))
```

Comment:
If the current style is Emacs this function could be a no-op.
In `spacemacs//magit-evil-magit-bindings` we can call again `git/init-evil-magit` if the passed style is `vim` or `hybrid` and somehow we find a way to detect if `evil-magit` is loaded (`feature-p` ?).

Also `hybrid` should respect the variable `hybrid-mode-enable-evilified-state`, if it is not turned on then the user probably does not want `evil-magit`.